### DISCUSSION TITLE
trace cross thread problem when use forkJoinPool/forkJoinTask

### DISCUSSION THREAD

**Discussion by Yelijah** [apache/skywalking#9808] [Q&A] [answered]
How to fix trace cross thread problem when use forkjoinPool/forkjoinTask? The[ official doc](https://skywalking.apache.org/docs/skywalking-java/v8.11.0/en/setup/service-agent/java-agent/application-toolkit-trace-cross-thread/) didn't mention it.

**Comment by wu-sheng** [comment, marked as answer]
There is no API to support that directly.

All x-thread APIs are calling kernel APIs in the agent, https://skywalking.apache.org/docs/skywalking-java/v8.11.0/en/setup/service-agent/java-agent/java-plugin-development-guide/#contextsnapshot.

You either consider learning kernel APIs to contribute a new toolkit APIs for these or learning OpenTracing(opentracing.io) adaptor(https://skywalking.apache.org/docs/skywalking-java/v8.11.0/en/setup/service-agent/java-agent/opentracing/)

Notice, OpenTracing 0.30.0 APIs are supported, but the project is archived(not maintained anymore). You have to take the risk, no one would response if there is a bug.

**Reply by wu-sheng** [reply]
Update in half of year later, **This has been added through https://github.com/apache/skywalking-java/pull/505**

NOTICE, this is how the community works. The thing would be done when someone shows up to help. Get yourself working on the thing.
